aws_embedded_metrics/logger/metrics_context.py

```diff
@@ -168,6 +178,16 @@ def create_copy_with_context(self) -> "MetricsContext":
             self.namespace, new_properties, new_dimensions, new_default_dimensions
         )
 
+    def create_copy_with_context_with_dimensions(self) -> "MetricsContext":
+        """
+        Creates a deep copy of the context excluding metrics.
+        Dimensions added with put_dimension will be copied, this helps reuse of dimension sets.
```

Comment:
Won't dimensions added with `set_dimensions` also be copied?

Reply:
Make sense. Will address the ambigous part.

Reply:
Also can you adjust "...this helps reuse of dimension sets" to "this helps with the reuse of dimension sets" or "this helps reuse dimension sets".

Reply:
Solved.